### DISCUSSION TITLE
Disabled rules not working

### DISCUSSION THREAD

**Discussion by wtl06639** [Security-Onion-Solutions/securityonion#3151] [Unsupported Versions] [answered]
I am trying to disable overly active rules per the documentation but they do not seem to take.  I am running a distributed environment with a manager and heavynodes. I have updated the global.sls file with the following I have even tried adding it to the minion sensor files but no change:

idstools:
  sids:
    disabled:
      - 38747
      - 38748
      - 38749
      - 38750
  
Per other discussion posts I have also adjusted the redis max memory to 3248 

In the logstash  log file I am seeing the following: 

[2021-02-26T06:40:18,479][WARN ][logstash.outputs.redis   ] Failed to send backlog of events to Redis {:identity=>"redis://@server:6379/0 list:logstash:unpars
tion=>#<Redis::CommandError: OOM command not allowed when used memory > 'maxmemory'.>, :backtrace=>["/usr/share/logstash/vendor/bundle/jruby/2.5.0/gems/redis-4.2
s/client.rb:127:in `call'", "/usr/share/logstash/vendor/bundle/jruby/2.5.0/gems/redis-4.2.2/lib/redis.rb:1158:in `block in rpush'", "/usr/share/logstash/vendor/b
/2.5.0/gems/redis-4.2.2/lib/redis.rb:69:in `block in synchronize'", "uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/monitor.rb:235:in `mon_synchronize'", "
logstash/vendor/bundle/jruby/2.5.0/gems/redis-4.2.2/lib/redis.rb:69:in `synchronize'", "/usr/share/logstash/vendor/bundle/jruby/2.5.0/gems/redis-4.2.2/lib/redis.
`rpush'", "/usr/share/logstash/vendor/bundle/jruby/2.5.0/gems/logstash-output-redis-5.0.0/lib/logstash/outputs/redis.rb:152:in `flush'", "/usr/share/logstash/ven
jruby/2.5.0/gems/stud-0.0.23/lib/stud/buffer.rb:221:in `block in buffer_flush'", "org/jruby/RubyHash.java:1415:in `each'", "/usr/share/logstash/vendor/bundle/jru
ms/stud-0.0.23/lib/stud/buffer.rb:216:in `buffer_flush'", "/usr/share/logstash/vendor/bundle/jruby/2.5.0/gems/stud-0.0.23/lib/stud/buffer.rb:159:in `buffer_recei
/share/logstash/vendor/bundle/jruby/2.5.0/gems/logstash-output-redis-5.0.0/lib/logstash/outputs/redis.rb:209:in `send_to_redis'", "/usr/share/logstash/vendor/bun
.5.0/gems/logstash-codec-json-3.0.5/lib/logstash/codecs/json.rb:42:in `encode'", "/usr/share/logstash/logstash-core/lib/logstash/codecs/delegator.rb:48:in `block
", "org/logstash/instrument/metrics/AbstractSimpleMetricExt.java:65:in `time'", "org/logstash/instrument/metrics/AbstractNamespacedMetricExt.java:64:in `time'",
/logstash/logstash-core/lib/logstash/codecs/delegator.rb:47:in `encode'", "/usr/share/logstash/vendor/bundle/jruby/2.5.0/gems/logstash-output-redis-5.0.0/lib/log
ts/redis.rb:123:in `receive'", "/usr/share/logstash/logstash-core/lib/logstash/outputs/base.rb:105:in `block in multi_receive'", "org/jruby/RubyArray.java:1809:i
"/usr/share/logstash/logstash-core/lib/logstash/outputs/base.rb:105:in `multi_receive'", "org/logstash/config/ir/compiler/OutputStrategyExt.java:138:in `multi_re
rg/logstash/config/ir/compiler/AbstractOutputDelegatorExt.java:121:in `multi_receive'", "/usr/share/logstash/logstash-core/lib/logstash/java_pipeline.rb:295:in `
art_workers'"]}

My sensors are filling up TB or space in hours an I need to reduce the amount of alerts coming through otherwise it will just keep maxing out and stop generating new alerts. 

I also found I can not run the 'salt <sensorname>_<role> state.apply idstools as it shows in the documentation. I just get the following error:

sensor_heavynode:
----------
          ID: idstools_state_not_allowed
    Function: test.fail_without_changes
      Result: False
     Comment: Failure!
     Started: 13:06:04.509038
    Duration: 4.741 ms
     Changes:

Summary for sensor_heavynode
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
Total run time:   4.741 ms
ERROR: Minions returned with non-zero exit code

Any idea why I can not get this to work? do i need to up the redis max memory? and if so what would be acceptable ?

**Comment by TOoSmOotH** [comment, marked as answer]
The error you are getting is your Redis queue is full. I would check logstash and elasticsearch on your heavy node to make sure it is ingesting data still. 

IDSTools only runs on the manager. That is why you see that error. 

If I had to make a guess to the root cause here, it sounds like your heavy node can't handle the amount of traffic you are throwing at it.

**Reply by wtl06639** [reply]
I have removed one of the interfaces that having traffic come in. That should cut the traffic in half. Still seems to be filling up quite quick. Can fill up about 1.5 TB in several hours.  It worked fine when we were running ubuntu 16.04 version before. I want to disable alerts but the commands per the documentation dont see to take. That may help reduce the logging. any idea how to resolve that?
